components/ca-import.js

```diff
@@ -0,0 +1,727 @@
+            }
+        }
+
+        /**
+         * Gets first row
+         * @returns {string} firstrow seperated by pipe
+         */
+        get firstRow() {
+            return (this.getAttribute('first-row') || '').split('|');
+        }
+
+        /**
+         * sets rawFile value
+         * @param {array} values for firstrow
+         */
+        set firstRow(values) {
+            // eslint-disable-next-line no-param-reassign
+            values = Array.isArray(values) ? values.join('|') : values;
+
+            if (values === '') {
+                this.removeAttribute('first-row');
+            } else {
+                this.setAttribute('first-row', values);
+            }
+        }
+
+
+        /**
+         * Gets hasData attribute
+         * @returns {string} hasData attribute
+         */
+        get hasData() {
+            return this.getAttribute('has-data');
+        }
+
+        /**
+         * Sets hasData attribute
+         * @param {boolean} value, boolean
+         */
+        set hasData(value) {
+            this.setAttribute('has-data', value === true);
+        }
+
+        /**
+         * Gets hasErrors attribute
+         * @param {string} hasErrors attribute
+         */
+        get hasErrors() {
+            return this.getAttribute('has-errors');
+        }
+
+        /**
+         * Sets hasErrors attribute
+         * @param {boolean} value, boolean
+         */
+        set hasErrors(value) {
+            this.setAttribute('has-errors', value === true);
+        }
+
+        /**
+         * Gets import type
+         * @param {string} type of import
+         */
+        get type() {
+            return this.getAttribute('type') || 'csv';
+        }
+
+        /**
+         * Sets type attribute
+         * @param {string} value - type of import
+         */
+        set type(value) {
+            // ensure its always csv or xlsx
+            // eslint-disable-next-line no-param-reassign
+            value = (value !== 'xlsx') ? value = 'csv' : 'xlsx';
+
+            this.setAttribute('type', value);
+            this.input.setAttribute('accept', `.${value}`);
+            this.link.setAttribute('download', `template.${value}`);
+        }
+
+        /**
+         * Gets convert url
+         * @param {string} convert url
+         */
+        get convertUrl() {
+            return this.getAttribute('convert-url') || '/ca-import';
+        }
+
+        /**
+         * Sets convertUrl
+         * @param {string} value, defaults to ca-import
+         */
+        set convertUrl(value = '/ca-import') {
+            this.setAttribute('convert-url', value);
+            this.form.action = value;
+        }
+
+        /**
+         * Gets end point that creates excel file from headers - only required for type=xls
+         * @returns {string} url
+         */
+        get templateConversionUrl() {
+            return this.getAttribute('template-conversion-url') || '/excel-download';
```

Comment:
I guess these routes will want to be part of the boilerplate server code we create.